synapse/handlers/room.py

```diff
@@ -469,13 +469,17 @@ async def clone_existing_room(
             (EventTypes.PowerLevels, ""),
         ]
 
-        # If the old room was a space, copy over the room type and the rooms in
-        # the space.
+        # Copy the room type as per MSC3818 https://github.com/matrix-org/matrix-spec-proposals/pull/3818.
+        if old_room_create_event.content.get(EventContentFields.ROOM_TYPE) is not None:
```

Comment:
Can we pull the field out once and re-use it in the two if-statements please.

```python
room_type = old_room_create_event.content.get(EventContentFields.ROOM_TYPE)
if room_type is not None:
    ...
if room_type == RoomTypes.SPACE:
    ...
```

Reply:
(Can likely indent the second if-statement too, but that's not super important!)